plugins/org.eclipse.n4js.flowgraphs/src/org/eclipse/n4js/flowgraphs/analysers/UsedBeforeDeclaredAnalyser2.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.flowgraphs.analysers;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.eclipse.n4js.flowgraphs.analyses.FastFlowVisitor;
+import org.eclipse.n4js.n4JS.ControlFlowElement;
+import org.eclipse.n4js.n4JS.IdentifierRef;
+import org.eclipse.n4js.n4JS.VariableDeclaration;
+
+/**
+ *
+ */
+public class UsedBeforeDeclaredAnalyser2 extends FastFlowVisitor {
```

Comment:
Why do you have this class `UsedBeforeDeclaredAnalyser2`?

Reply:
This implementation is faster and simpler because it is built upon the FastFlowVisitor. The other implementation will be deleted once one of them is actually used.